File: Day323/transformToPrime.java
//{ Driver Code Starts
    import java.io.*;
    import java.util.*;
    
     class GFG {
        public static void main(String[] args) throws IOException {
            BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
            int testcases = Integer.parseInt(br.readLine());
            while (testcases-- > 0) {
                int n = Integer.parseInt(br.readLine());
                String line1 = br.readLine();
                String[] a1 = line1.trim().split("\\s+");
                int a[] = new int[n];
                for (int i = 0; i < n; i++) {
                    a[i] = Integer.parseInt(a1[i]);
                }
                Solution ob = new Solution();
                int ans=ob.minNumber(a,n);
                System.out.println(ans);
            }
        }
    }
    
    // } Driver Code Ends
    
    
    //User function Template for Java
    
    
    class Solution
    {
        public boolean isPrime(int n) {
            for (int i=2; i<=Math.sqrt(n); i++) {
                if (n%i == 0) {
                    return false;
                }
            }
            return true;
        }
        public int minNumber(int arr[], int N)
        {
            int sum = 0;
            for (int i=0; i<N; i++) {
                sum += arr[i];
            }
            if (sum == 0) {
                return 0;
            }
            if (sum == 1) {
                return 1;
            }
            int temp = sum;
            while (!isPrime(sum)) {
                sum++;
            }
            return sum - temp;
        }
    }
    

File: Day323/transformToPrime.go
package main

import (
	"fmt"
	"math"
)

func isPrime(n int) bool {
	for i:=2; i<= int(math.Sqrt(float64(n))); i++ {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func minNumber(arr []int) int {
	sum := 0
	for _,v := range arr {
		sum += v
	}
	if sum == 0 {
		return 0
	}
	if sum == 1 {
		return 1
	}
	temp := sum
	for !isPrime(sum) {
		sum++
	}
	return sum - temp
}

func main() {
	arr1 := []int{2, 4, 6, 8, 12}
	arr2 := []int{1, 5, 7}
	arr3 := []int{1, 0, 0}
	fmt.Println(minNumber(arr1))
	fmt.Println(minNumber(arr2))
	fmt.Println(minNumber(arr3))
}